dashboard/app/views/levels/_dialog.html.haml

```diff
@@ -125,20 +125,38 @@
     adjustedScroll = true;
   }
 
+  // TODO(dave): Dashboard shouldn't be reaching into the internal implementation of
+  // individual levels. Instead levels should call appOptions.onAttempt.
   $('.submitButton').click(function () {
     var submitButton = $('.submitButton');
     if (submitButton.attr('disabled')) {
       return;
     }
 
+    // Avoid multiple simultaneous submissions.
+    submitButton.attr('disabled', true);
+
+    var onCompleteNoRedirect = function() {
+      $('.submitButton').attr('disabled', false);
+    };
+
+    processResults(onCompleteNoRedirect);
+  });
+
+  /**
+   * Process the solution to the puzzle submitted by the user.
+   * @param {Function} onCompleteNoRedirect Optional callback function to call if
```

Comment:
Would it simplify things conceptually to just make this onComplete, pass it the lastServerResponse, and let the caller decide to gate on lastServerResponse.nextRedirect?

Might not matter if we're planning on moving this logic elsewhere in the nearish future anyways.